client/ui/components/forms/fields/AddonSelectField.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { PropTypes as pt } from 'react';
+
+import Chip from 'material-ui/Chip';
+import MenuItem from 'material-ui/MenuItem';
+import SelectField from 'material-ui/SelectField';
+
+
+const style = {
+  addonChip: {
+    marginRight: '5px',
+  },
+  addonChips: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+};
+
+
+class AddonSelectField extends React.Component {
+  static propTypes = {
+    addons: pt.array.isRequired,
+    floatingLabelText: pt.string.isRequired,
+    input: pt.object.isRequired,
+    name: pt.string.isRequired,
+  };
+
+  constructor() {
+    super();
+
+    // Bind `this` to event handlers and render functions
```

Comment:
Don't worry about it for this PR, but https://github.com/jayphelps/core-decorators.js#autobind is a neater way to handle this in a future refactoring.